src/lib/Modal.svelte

```diff
@@ -0,0 +1,31 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+
+	let mounted = false;
+	onMount(() => {
+		mounted = true;
+	});
+	let dialog: HTMLDialogElement;
+	export let open: boolean;
+
+	$: if (open && mounted) {
```

Comment:
It looks like you are always wrapping this component in an if statement, cant this just be run onMont then?